src/envoy/http/service_control/client_cache.h

```diff
@@ -49,6 +52,11 @@ class ClientCache : public Envoy::Logger::Loggable<Envoy::Logger::Id::filter> {
       const ::google::api::servicecontrol::v1::CheckRequest& request,
       Envoy::Tracing::Span& parent_span, CheckDoneFunc on_done);
 
+  // Ownership of CheckResponse is passed to this function.
+  // The function will always call CheckDoneFunc.
+  void handleCheckResponse(const Status& http_status, CheckResponse* response,
```

Comment:
Technically, this should not be part of the public interface. But I need to put it here for unit testing. Is this ok @qiwzhang?

I could move these into a separate class, but that seems like overkill for just 1 function.

Reply:
it is better not to pollute the public interfaces. 
see if we can move into "private" and use "friend class" to allow test class to access it.  
need to move test class out of annoymous name space.

Reply:
Done with friend class.